Assets/Src/Scripts/SeedAvatar/Input/WebCamSource.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Events;
+using Debug = UnityEngine.Debug;
+
+namespace SeedAvatar {
+  public enum ResolutionOptions {
+    LOW = 0,
+    MIDDLE = 1,
+    HIGH = 2,
+  }
+
+  public enum FpsOptions {
+    LOW = 0,
+    HIGH = 1,
+  }
+
+  public enum FlipOptions {
+    NORMAL = 0,
+    MIRRORING = 1,
+  }
+
+  public class WebCamSource : MonoBehaviour {
+    public Dropdown CameraDropdown;
+    public Dropdown ResolutionDropdown;
+    public Dropdown FpsDropdown;
+    public Dropdown FlipDropdown;
+    public string DeviceName;
+    public int Fps = 30;
+    public bool Flip = false;
+    [SerializeField] private RenderTexture _buffer;
+    private WebCamTexture _webCam;
+    private int _width = 1280;
+    private int _height = 720;
+    private static readonly Dictionary<int, string> _resolutionDict =
+    new Dictionary<int, string> {
+      { (int)ResolutionOptions.LOW , "1280x720" },
+      { (int)ResolutionOptions.MIDDLE, "1600x900" },
+      { (int)ResolutionOptions.HIGH , "1920x1080" },
+    };
+    private static readonly Dictionary<int, string> _fpsDict =
+    new Dictionary<int, string> {
+      { (int)FpsOptions.LOW , "30Hz" },
+      { (int)FpsOptions.HIGH , "60Hz" },
+    };
+    private static readonly Dictionary<int, string> _flipDict =
+    new Dictionary<int, string> {
+      { (int)FlipOptions.NORMAL , "Normal" },
+      { (int)FlipOptions.MIRRORING , "Mirroring" },
+    };
+    private Dictionary<int, string> _devicesDict = new Dictionary<int, string> { };
+
+    // Start button event.
+    public void Play() {
+      if (_webCam == null) {
+        _webCam = new WebCamTexture(DeviceName, _width, _height, Fps);
+        _webCam.Play();
+      }
+    }
+
+    // Stop button event.
+    public void Stop() {
+      if (_webCam != null) {
+        _webCam.Stop();
+        _webCam = null;
+      }
+    }
+
+    // DropdownChange event.
+    public void OnCameraDropdownChange(int value) {
+      if (_devicesDict[value] != null) {
```

Comment:
This is not the correct way to check if a dictionary contains a key. Typically dict[key] will throw an exception if the key doesn't exist.

Please use `Dictionary.ContainsKey` or `Dictionary.TryGetValue` to check if the key exists.